uni/lib/view/home/widgets/main_cards_list.dart

```diff
@@ -52,61 +52,95 @@ class MainCardsListState extends State<MainCardsList> {
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      body: BackButtonExitWrapper(
-        child: SizedBox(
-          height: MediaQuery.of(context).size.height,
-          child: isEditing
-              ? ReorderableListView(
-                  onReorder: reorderCard,
-                  header: createTopBar(context),
-                  children: favoriteCardsFromTypes(
-                    widget.favoriteCardTypes,
-                    context,
-                  ),
-                )
-              : ListView(
-                  children: <Widget>[
-                    createTopBar(context),
-                    ...favoriteCardsFromTypes(
+    return WillPopScope(
+      onWillPop: () async {
+        if (isEditing) {
+          setState(() {
+            isEditing = false;
+          });
+          return false;
+        }
+        return true;
+      },
+      child: Scaffold(
+        body: BackButtonExitWrapper(
+          child: SizedBox(
+            height: MediaQuery.of(context).size.height,
+            child: isEditing
+                ? ReorderableListView(
+                    onReorder: (oldIndex, newIndex) {
+                      setState(() {
+                        if (oldIndex < newIndex) {
+                          newIndex -= 1;
+                        }
```

Comment:
What does this line do?

Reply:
The source code occasionally failed to update the Index reliably in this instance, so these lines were added as a safeguard to ensure its proper updating.